dot/state/service_test.go

```diff
@@ -128,3 +129,72 @@ func TestService_BlockTree(t *testing.T) {
 		t.Fatalf("Fail: got %s expected %s", stateA.Block.BestBlockHash(), stateB.Block.BestBlockHash())
 	}
 }
+func Test_ServicePruneStorage(t *testing.T) {
+	testDir := utils.NewTestDir(t)
+	defer utils.RemoveTestDir(t)
+
+	serv := NewService(testDir, log.LvlTrace)
+	serv.UseMemDB()
+
+	genesisHeader, err := types.NewHeader(common.NewHash([]byte{0}), big.NewInt(0), trie.EmptyHash, trie.EmptyHash, [][]byte{})
+	require.Nil(t, err)
+
+	genesisData := new(genesis.Data)
+
+	tr := trie.NewEmptyTrie()
+	err = serv.Initialize(genesisData, genesisHeader, tr, firstEpochInfo)
+	require.Nil(t, err)
+
+	err = serv.Start()
+	require.Nil(t, err)
+
+	serv.Block = newTestBlockState(t, testGenesisHeader)
+	serv.Storage = newTestStorageState(t)
+	ch := make(chan *types.Header, 3)
+	id, err := serv.Block.RegisterFinalizedChannel(ch)
+	require.NoError(t, err)
+
+	defer serv.Block.UnregisterFinalizedChannel(id)
+
+	chain, _ := AddBlocksToState(t, serv.Block, 3)
+
+	type prunedBlock struct {
+		hash  common.Hash
+		dbKey []byte
+	}
+
+	var prunedArr []prunedBlock
+	for i, b := range chain {
+		emptyTrie := trie.NewEmptyTrie()
+		chainHash := b.Hash()
+		err := emptyTrie.Put(chainHash[:], chainHash[:])
+		require.NoError(t, err)
+
+		// Update the block in StorageState manually.
+		serv.Storage.tries[b.Hash()] = emptyTrie
+		err = serv.Storage.StoreInDB(b.Hash())
+		require.NoError(t, err)
+
+		if i == len(chain)-1 {
+			serv.Block.SetFinalizedHash(b.Hash(), 0, 0)
+		} else {
+			rootHash, err := emptyTrie.Hash()
+			require.NoError(t, err)
+			prunedArr = append(prunedArr, prunedBlock{hash: b.Hash(), dbKey: rootHash[:]})
+		}
+	}
+
+	time.Sleep(10 * time.Second)
```

Comment:
this seems long? is it necessary to be this long?

Reply:
Updated it to one sec.